cvat-ui/src/components/move-task-modal/move-task-modal.tsx

```diff
@@ -0,0 +1,158 @@
+
+const core = getCore();
+
+export default function MoveTaskModal(): JSX.Element {
+    const visible = useSelector((state: CombinedState) => state.tasks.moveTask.modalVisible);
+    const task = useSelector((state: CombinedState) => {
+        const [taskInstance] = state.tasks.current.filter((_task) => _task.instance.id === state.tasks.moveTask.taskId);
+        return taskInstance?.instance;
+    });
+    const taskUpdating = useSelector((state: CombinedState) => state.tasks.updating);
+    const dispatch = useDispatch();
+
+    const [projectId, setProjectId] = useState<number | null>(null);
+    const [project, setProject] = useState<any>(null);
+    const [values, setValues] = useState<{ [key: string]: LabelMapperItemValue }>({});
+
+    const initValues = (): void => {
+        if (task) {
+            const labelValues: { [key: string]: LabelMapperItemValue } = {};
+            task.labels.forEach((label: any) => {
+                labelValues[label.id] = {
+                    labelId: label.id,
+                    newLabelName: null,
+                    clearAtrributes: true,
+                };
+            });
+            setValues(labelValues);
+        }
+    };
+
+    const onCancel = (): void => {
+        dispatch(closeMoveTaskModal());
+        initValues();
+        setProject(null);
+        setProjectId(null);
+    };
+
+    const submitMove = async (): Promise<void> => {
+        if (!projectId) {
+            notification.error({
+                message: 'Project not selected',
+            });
+            return;
+        }
+        if (!Object.values(values).every((_value) => _value.newLabelName !== null)) {
+            notification.error({
+                message: 'Not all labels mapped',
+                description: 'Please choose any action to not mapped labels first',
+            });
+            return;
+        }
+        dispatch(
+            moveTaskToProjectAsync(
+                task,
+                projectId,
+                Object.values(values).map((value) => ({
+                    label_id: value.labelId,
+                    new_label_name: value.newLabelName,
+                    clear_attributes: value.clearAtrributes,
+                })),
+            ),
+        );
+        onCancel();
+    };
+
+    useEffect(() => {
+        if (projectId) {
+            core.projects.get({ id: projectId }).then((_project: any) => {
+                if (projectId) {
+                    setProject(_project[0]);
+                    const { labels } = _project[0];
+                    const labelValues: { [key: string]: LabelMapperItemValue } = {};
+                    Object.entries(values).forEach(([id, label]) => {
+                        const [autoNewLabel] = labels.filter((_label: any) => (
+                            _label.name === task.labels.filter((_taskLabel: any) => (
+                                _taskLabel.id === label.labelId
+                            ))[0].name
```

Comment:
The structure looks complex. Can we put internal ``filter`` outside before?

Reply:
done